Statistics/NumberCruncher.cpp

```diff
@@ -224,7 +224,36 @@ void NumberCruncher::merge_rotation_executor_stats(QList<QList<ExecutorOutcome*>
         list.append(rotation_executor->get_list_of_executor_outcomes());
 }
 
-void NumberCruncher::calculate_stat_weights(QList<ScaleResult*>& list) {
+void NumberCruncher::calculate_stat_weights_for_tps(QList<ScaleResult*>& tps_list) {
+    QMutexLocker lock(&mutex);
+
+    QMap<SimOption::Name, double> tps_per_option;
+
+    QMap<SimOption::Name, QVector<ClassStatistics*>>::const_iterator it = class_stats.constBegin();
+    auto end = class_stats.constEnd();
+    while (it != end) {
+        tps_per_option.insert(it.key(), get_tps_for_option(it.key()));
+        ++it;
+    }
+
+    check(tps_per_option.contains(SimOption::Name::NoScale), "Missing baseline NoScale statistics");
+
+    double base_value_tps = tps_per_option.take(SimOption::Name::NoScale);
+
+    QMap<SimOption::Name, double>::const_iterator tps_it = tps_per_option.constBegin();
+    while (tps_it != tps_per_option.constEnd()) {
+      double absolute_diff = tps_it.value() - base_value_tps;
+      double relative_diff = absolute_diff / base_value_tps;
+
+      double standard_deviation = get_standard_deviation_for_option(tps_it.key());
+      double confidence_interval = get_confidence_interval_for_option(tps_it.key(), standard_deviation);
```

Comment:
Indentation is lower than expected in this project (and as formatted by clang-format).